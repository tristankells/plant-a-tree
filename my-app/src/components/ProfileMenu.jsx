import React, { Component } from "react";
import "../style/ProfileMenu.css";
import ClearIcon from "@material-ui/icons/Clear";

class ProfileMenu extends Component {
  render() {
    var visibility = "hide";

    if (this.props.menuVisibility) {
      visibility = "show";
    }

    return (
      <div id="profileMenu" className={visibility}>
        <ClearIcon className="slider-close-icon" onMouseDown={this.props.handleMouseDown} />
        {this.props.children}

        <div className="account-section__header">ACCOUNT</div>

        <div className="login-section__header"><b>Signup</b></div>

          <label>
          Username:
            <input
              name="username"
              type="text"
              checked={this.props.username}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            Password:
            <input
              name="password"
              type="password"
              value={this.props.password}
              onChange={this.props.handleInputChange} />
          </label><br/><br/>
          <label>
            Full name:
            <input
              name="fullName"
              type="text"
              value={this.props.fullname}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            Email:
            <input
              name="email"
              type="text"
              value={this.props.email}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            Street & no:
            <input
              name="address"
              type="text"
              value={this.props.address}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            Suburb:
            <input
              name="town"
              type="text"
              value={this.props.town}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            City:
            <input
              name="city"
              type="text"
              value={this.props.city}
              onChange={this.props.handleInputChange} />
          </label><br/>
          <label>
            Postcode:
            <input
              name="postcode"
              type="text"
              value={this.props.postcode}
              onChange={this.props.handleInputChange} />
          </label><br/>
          
          <button className="profile-submit-button" type="submit" value="Sign Up" onClick={this.props.signUpSubmit}>Sign Up</button>
        
      </div>
    );
  }
}
export default ProfileMenu;