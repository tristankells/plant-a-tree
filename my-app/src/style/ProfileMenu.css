#profileMenu {
  width: 70vw;
  max-width: 400px;
  height: 100vh;
  padding: 20px;
  background-image: linear-gradient(-90deg, rgb(43, 66, 31), #182A0F);
  color: white;
  border-radius: 5px 0px 0px 5px;
  position: fixed;
  top: 0;
  right: 0;
  transition: transform 0.3s cubic-bezier(0, 0.52, 0, 1);
  overflow: scroll;
  z-index: 1000;
  opacity: 0.92;
}

#profileMenu.hide {
  transform: translate3d(+100vw, 0, 0);
}

#profileMenu.show {
  transform: translate3d(0vw, 0, 0);
  overflow: hidden;
}

#profileMenu h2 a {
  color: #333;
  margin-left: 15px;
  text-decoration: none;
}

#profileMenu h2 a:hover {
  text-decoration: underline;
}

#profileMenu input {
  border: 0px;
  border-radius: 20px;
  height: 25px;
  margin: 5px;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
  float: right;
}

#profileMenu input:focus{
  outline-width: 0;
}

#profileMenu label {
  font-size: 12px;
  display: inline-block;
  float: right;
  width: 250px;
  text-align: right;
}

.account-section__header{
  font-family: Arial, Helvetica, sans-serif;
  font-size: 20px;
  font-weight: bolder;
  margin-bottom: 30px;
  margin-top: 20px;
  letter-spacing: 5px;
  
}

.login-section__header{
  margin: 50px 0px 20px 0px;
}

.signup-section__header{
  margin: 50px 0px 20px 0px;
}

.profile-submit-button {
  margin: 20px 10px 20px 0px;
  padding: 10px 30px 10px 30px;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
  border-style: solid;
  border-color: white;
  border-width: 1px;
  background-color: transparent;
  border-radius: 25px;
  color: white;
}

.submit-button:hover{
  cursor: pointer;
  background-color: white;
  color: black;
}

/* Slider close icon */
.slider-close-icon {
  margin-top: 25px;
  margin-bottom: 0px;
  padding: 1px;
  border-radius: 50px;
  padding: 5px;
}

.slider-close-icon:hover{ 
  color: white;
  background-color: rgb(48, 61, 51);
  border-radius: 50px;
  cursor: pointer;
}