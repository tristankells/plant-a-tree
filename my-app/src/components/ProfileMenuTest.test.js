import React from "react";
import { shallow } from "enzyme";

import ProfileMenu from "./ProfileMenu";

describe("ProfileMenu", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ProfileMenu debug />);

    expect(component).toMatchSnapshot();
  });
});
