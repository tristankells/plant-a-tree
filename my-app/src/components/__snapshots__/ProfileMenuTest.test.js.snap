// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu should render correctly in "debug" mode 1`] = `
<div
  className="hide"
  id="profileMenu"
>
  <pure(ClearIcon)
    className="slider-close-icon"
  />
  <div
    className="account-section__header"
  >
    ACCOUNT
  </div>
  <div
    className="login-section__header"
  >
    <b>
      Login / Signup
    </b>
  </div>
  <label>
    Username:
    <input
      name="username"
      type="text"
    />
  </label>
  <br />
  <label>
    Password:
    <input
      name="password"
      type="password"
    />
  </label>
  <br />
  <button
    className="profile-submit-button"
    type="submit"
    value="Log In"
  >
    Log In
  </button>
  <button
    className="profile-submit-button"
    type="submit"
    value="Sign Up"
  >
    Sign Up
  </button>
</div>
`;
